net/redisclient.go

```diff
@@ -298,7 +299,7 @@ func (r *RedisRingClient) startUpdater(ctx context.Context) {
 		case <-ticker.C:
 		}
 
-		addrs := r.options.AddrUpdater()
+		addrs, _ := r.options.AddrUpdater()
```

Comment:
Here we need to check error and if non nil log.Error and continue so stop further processing to not pass the empty list.